Source/Particles/Collision/BinaryCollision/ParticleCreationFunc.H

```diff
@@ -198,10 +198,12 @@ public:
                 }
 
                 // Remove p_pair_reaction_weight[i] from the colliding particles' weights
-                BinaryCollisionUtils::remove_weight_from_colliding_particle(
-                    w1[p_pair_indices_1[i]], idcpu1[p_pair_indices_1[i]], p_pair_reaction_weight[i]);
-                BinaryCollisionUtils::remove_weight_from_colliding_particle(
-                    w2[p_pair_indices_2[i]], idcpu2[p_pair_indices_2[i]], p_pair_reaction_weight[i]);
+                if (t_collision_type == CollisionType::DSMC) { // performed locally for fusion
+                    BinaryCollisionUtils::remove_weight_from_colliding_particle(
+                        w1[p_pair_indices_1[i]], idcpu1[p_pair_indices_1[i]], p_pair_reaction_weight[i]);
+                    BinaryCollisionUtils::remove_weight_from_colliding_particle(
+                        w2[p_pair_indices_2[i]], idcpu2[p_pair_indices_2[i]], p_pair_reaction_weight[i]);
+                }
```

Comment:
The DSMC routine doesn't use `ParticleCreationFunc.H`. This is just some unfortunate file organization and naming since the nuclear fusion routine was added before DSMC. For DSMC the file `SplitAndScatterFunc.H` inside the DSMC folder handles particle splitting and scattering.
It would be a good topic to add for the hackathon to move this into the NuclearFusion folder.

The inline comment above this line can also be removed but I can't do that as part of the suggestion.
```suggestion
```